TCAT/Base/AppDelegate.swift

```diff
@@ -100,20 +87,35 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
     // MARK: - Helper Functions
 
+    /// Initializes the UserDefaults values if not present
+    private func initializeUserDefaults() {
+        for (key, defaultValue) in userDataInits {
+            if userDefaults.value(forKey: key) == nil {
+                if key == Constants.UserDefaults.favorites && sharedUserDefaults?.value(forKey: key) == nil {
+                    sharedUserDefaults?.set(defaultValue, forKey: key)
+                } else {
+                    userDefaults.set(defaultValue, forKey: key)
+                }
+            } else if key == Constants.UserDefaults.favorites && sharedUserDefaults?.value(forKey: key) == nil {
+                sharedUserDefaults?.set(userDefaults.value(forKey: key), forKey: key)
+            }
+        }
+    }
+
     /// Creates and sets a unique identifier. If the device identifier changes, updates it.
-    func setupUniqueIdentifier() {
+    private func setupUniqueIdentifier() {
         if let uid = UIDevice.current.identifierForVendor?.uuidString,
-            uid != sharedUserDefaults?.string(forKey: Constants.UserDefaults.uid) {
+           uid != sharedUserDefaults?.string(forKey: Constants.UserDefaults.uid) {
```

Comment:
Maybe using a guard statement here (i.e. guard let uid = UIDevice.current.identifierForVendor?.uuidString) for better readability? But not a big issue so up to you!